master/buildbot/reporters/stash.py

```diff
@@ -32,25 +34,47 @@ class StashStatusPush(http.HttpStatusPushBase):
     name = "StashStatusPush"
 
     @defer.inlineCallbacks
-    def reconfigService(self, base_url, user, password, **kwargs):
+    def reconfigService(self, base_url, user, password, key=None, statusName=None,
+                        startDescription=None, endDescription=None,
+                        verbose=False, **kwargs):
         yield http.HttpStatusPushBase.reconfigService(self, **kwargs)
+        self.key = key or Interpolate('%(prop:buildername)s')
+        self.statusName = statusName
+        self.endDescription = endDescription
+        self.startDescription = startDescription
+        self.verbose = verbose
         self._http = yield httpclientservice.HTTPClientService.getService(
             self.master, base_url, auth=(user, password))
 
     @defer.inlineCallbacks
     def send(self, build):
+        props = Properties.fromDict(build['properties'])
         results = build['results']
         if build['complete']:
             status = STASH_SUCCESSFUL if results == SUCCESS else STASH_FAILED
+            description = self.endDescription
         else:
             status = STASH_INPROGRESS
+            description = self.startDescription
         for sourcestamp in build['buildset']['sourcestamps']:
             sha = sourcestamp['revision']
-            body = {'state': status, 'key': build[
-                'builder']['name'], 'url': build['url']}
+            key = yield props.render(self.key)
+            payload = {
+                'state': status,
+                'url': build['url'],
+                'key': key,
+            }
+            if description:
+                payload['description'] = yield props.render(description)
+            if self.statusName:
+                payload['name'] = yield props.render(self.statusName)
             response = yield self._http.post('/rest/build-status/1.0/commits/' + sha,
-                                            json=body)
-            if response.code != 204:
+                                             json=payload)
+            if response.code == 204:
+                if self.verbose:
+                    log.msg('Status "{status}" sent for {sha}.'.format(
```

Comment:
please use logger API, and structured logging for new code

    log.info('Status "{status}" sent for {sha}.',  status=status, sha=sha)

utils like txlogstash can then output that in json format